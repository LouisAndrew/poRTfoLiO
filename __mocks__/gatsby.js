const React = require('react')
const gatsby = jest.requireActual('gatsby')

const mockSocials = [
    { link: 'http://localhost', social_key: 'github' },
    { link: 'http://google.com', social_key: 'dribbble' },
]

const mockButtonText = 'Click me'
const mockContactContent = 'Content'
const mockContactHeadline = 'Headline'
const mockEmail = 'john@doe.com'

module.exports = {
    ...gatsby,
    graphql: jest.fn(),
    Link: jest.fn().mockImplementation(
        // these props are invalid for an `a` tag
        ({
            activeClassName,
            activeStyle,
            getProps,
            innerRef,
            ref,
            replace,
            to,
            ...rest
        }) =>
            React.createElement('a', {
                ...rest,
                href: to,
            })
    ),
    StaticQuery: jest.fn(),
    // can I mock return value here?
    useStaticQuery: jest.fn(() => ({
        socialData: {
            frontmatter: {
                social: mockSocials,
            },
        },
        emailData: {
            frontmatter: {
                contact: {
                    button_text: mockButtonText,
                    contact_content: mockContactContent,
                    contact_headline: mockContactHeadline,
                    email: mockEmail,
                },
            },
        },

        // forced to write the hero-page-query mock here.
        heroData: {
            frontmatter: {
                hero: {
                    hero_content: 'Hello, World!', // hard coded from hero.test.tsx
                    hero_headline: 'I am headline', // Also hard coded from hero.test.tsx
                },
            },
        },
        imgS: {
            frontmatter: {
                hero: {
                    hero_imgs: [
                        {
                            childImageSharp: {
                                fixed: {},
                            },
                        },
                    ],
                },
            },
        },
        imgM: {
            frontmatter: {
                hero: {
                    hero_imgs: [
                        {
                            childImageSharp: {
                                fixed: {},
                            },
                        },
                    ],
                },
            },
        },
        imgL: {
            frontmatter: {
                hero: {
                    hero_imgs: [
                        {
                            childImageSharp: {
                                fixed: {},
                            },
                        },
                    ],
                },
            },
        },
        imgXL: {
            frontmatter: {
                hero: {
                    hero_imgs: [
                        {
                            childImageSharp: {
                                fixed: {},
                            },
                        },
                    ],
                },
            },
        },
    })),
}
