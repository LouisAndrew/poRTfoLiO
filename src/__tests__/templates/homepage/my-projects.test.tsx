import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import { cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'

import MyProjects from 'templates/homepage/my-projects'
// import { nameToSlug } from 'templates/homepage/my-projects/my-projects'

import * as Gatsby from 'gatsby'

const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery')

const mockDesc = 'description'
const mockName = 'Sample Project'
const mockLabel = 'Label'

describe('MyProjects', () => {
    const Element = <MyProjects />

    afterEach(cleanup)
    beforeEach(() => {
        useStaticQuery.mockImplementationOnce(() => ({
            projects: {
                edges: [
                    {
                        node: {
                            frontmatter: {
                                preview_desc: mockDesc,
                                project_name: mockName,
                                project_screenshots: [
                                    {
                                        screenshot_label: mockLabel,
                                    },
                                ],
                            },
                        },
                    },
                ],
            },
            imgSQuery: {
                edges: [
                    {
                        node: {
                            frontmatter: {
                                project_screenshots: [
                                    {
                                        screenshot: {
                                            childImageSharp: {
                                                fluid: {},
                                            },
                                        },
                                    },
                                ],
                            },
                        },
                    },
                ],
            },
            imgMQuery: {
                edges: [
                    {
                        node: {
                            frontmatter: {
                                project_screenshots: [
                                    {
                                        screenshot: {
                                            childImageSharp: {
                                                fluid: {},
                                            },
                                        },
                                    },
                                ],
                            },
                        },
                    },
                ],
            },
            imgLQuery: {
                edges: [
                    {
                        node: {
                            frontmatter: {
                                project_screenshots: [
                                    {
                                        screenshot: {
                                            childImageSharp: {
                                                fluid: {},
                                            },
                                        },
                                    },
                                ],
                            },
                        },
                    },
                ],
            },
            imgXLQuery: {
                edges: [
                    {
                        node: {
                            frontmatter: {
                                project_screenshots: [
                                    {
                                        screenshot: {
                                            childImageSharp: {
                                                fluid: {},
                                            },
                                        },
                                    },
                                ],
                            },
                        },
                    },
                ],
            },
        }))
    })

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(Element, div)
    })

    it('matches snapshot', () => {
        const run = true

        expect(run).toBeTruthy()

        if (run) {
            const tree = renderer.create(Element).toJSON()
            expect(tree).toMatchSnapshot()
        }
    })
})
