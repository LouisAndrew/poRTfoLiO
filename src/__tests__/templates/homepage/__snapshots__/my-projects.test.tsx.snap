// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProjects matches snapshot 1`] = `
<div
  className="css-4cffwv"
>
  <div
    className="css-qdscj0"
  >
    <h2
      className="css-1hc7vq6"
    >
      My Projects
    </h2>
    <div
      className="css-arncl7"
    >
      <div
        className="css-i624kq"
      >
        <div
          className=" gatsby-image-wrapper"
          style={
            Object {
              "height": "100%",
              "maxHeight": null,
              "maxWidth": null,
              "overflow": "hidden",
              "position": "relative",
              "width": "100%",
            }
          }
        >
          <div
            aria-hidden={true}
            style={
              Object {
                "paddingBottom": "NaN%",
                "width": "100%",
              }
            }
          />
          <picture>
            <source
              media="(max-width: 48em) and (orientation: portrait)"
            />
            <source
              media="(max-width: 60em) and (orientation: landscape)"
            />
            <source
              media="(min-width: 122em)"
            />
            <source />
            <img
              alt=""
              loading="lazy"
              onLoad={[Function]}
              style={
                Object {
                  "height": "100%",
                  "left": 0,
                  "objectFit": "cover",
                  "objectPosition": "center",
                  "opacity": 0,
                  "position": "absolute",
                  "top": 0,
                  "transition": "opacity 500ms",
                  "width": "100%",
                }
              }
            />
          </picture>
          <noscript
            dangerouslySetInnerHTML={
              Object {
                "__html": "<picture><source media=\\"(max-width: 48em) and (orientation: portrait)\\" srcset=\\"undefined\\" /><source media=\\"(max-width: 60em) and (orientation: landscape)\\" srcset=\\"undefined\\" /><source media=\\"(min-width: 122em)\\" srcset=\\"undefined\\" /><source srcset=\\"undefined\\" /><img loading=\\"lazy\\" src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
              }
            }
          />
        </div>
      </div>
      <div
        className="css-o8s4w0"
      >
        <h2
          className="css-zrl3mw"
          data-testid="name"
        >
          Sample Project
        </h2>
        <p
          className="css-vurnku"
          data-testid="misc"
        >
          description
        </p>
        <a
          href="sample_project"
        >
          <button
            className="css-21tlyt"
          >
            View Project
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
